src/loki/search.cc

```diff
@@ -282,7 +300,11 @@ struct bin_handler_t {
         GraphId id = tile->id();
         id.set_id(node->edge_index() + (edge - start_edge));
         auto info = tile->edgeinfo(edge->edgeinfo_offset());
-
+        // calculate the heading of the snapped point to the shape for use in heading filter
+        size_t index = edge->forward() ? 0 : info.shape().size() - 2;
+        float angle =
+            tangent_angle(index, candidate.point, info.shape(),
+                          GetOffsetForHeading(edge->classification(), edge->use()), edge->forward());
```

Comment:
well this is unfortunate, but correct. we need to measure the tangent at the node for every edge. this means a bunch of tangent checks. if performance doesnt check out we shoudl think about putting the distance check first before the tangent... but then we have the fact that heading might need it... gah, lets see what performance numbers say before we complicate our lives..